Recruit-Training/Embedded-Recruit-Training/Core/Src/BlueLedToggleTask.c

```diff
@@ -9,8 +11,34 @@ void blueLedToggleTask(void const* arg)
 
     for (;;)
     {
-        osDelayUntil(prevWakeTime + 1);  //TODO: Replace 1 with time period for delay
+        osDelayUntil(prevWakeTime + BLUE_LED_TOGGLE_FREQ);  //TODO: Replace 1 with time period for delay
         //TODO: Check blue toggle flag and toggle blue LED
+        if(blueFlag == 1)
+        {
+            HAL_GPIO_TogglePin(GPIOA, LED_RED_Pin);
+        }
+        uint8_t blueLedVal = HAL_GPIO_ReadPin(GPIOA, LED_RED_Pin);
         //TODO: Send CAN message indicating current state of LED
+
+        if(osMutexAcquire(canMutex, 0) == osOK)
+        {
+            if(HAL_CAN_GetTxMailboxesFreeLevel(&hcan2) > 0)
+            {
+                uint8_t data[1] = {0};
+                uint32_t mailbox; 
+                CANTxHeader.StdId = BLUE_LED_STATUS_STDID;
+                CANTxHeader.DLC = 1;
+                
+                data[0] = !blueLedVal;
+
+                HAL_CAN_AddTxMessage(&hcan2, &CANTxHeader, &data, &mailbox);
+                osMutexRelease(canMutex);
+            }
+            osMutexRelease(canMutex); 
+        }
+        else
+        {
+            continue;
```

Comment:
You don't need this ```continue``` if you did want to use it you could change the if to check for ```!= osOK``` and put the continue in there and have all your after that (as the continue would take you back to the beginning of the loop